### ISSUE TITLE
error

### ISSUE THREAD

**Issue by galifrey35** [BSData/wh40k-7th-edition#638]
java.lang.IllegalArgumentException: Parameter 'directory' is not a directory

**Comment by HeWhoWatches** [comment]
Needs some additional information please:
Which version of BattleScribe and which platform / OS are you using?
What were you doing when this error occurred, for example was this error received when loading a particular catalogue?
What steps would we need to follow to try to reproduce the error please?

**Comment by Kangodo** [comment]
Probably the Blood Angels-one, is that correct? Because it should be gone now.
@galifrey35

**Comment by galifrey35** [comment]
i downloaded the free verirsion and i think the latest version 15 or what
ever and i was trying to add the data files went to manage data search for
files then went to some githup and one other place and got the same error
both times
i am using windows 7
i got the following error so here it is and the deatials
java.lang.IllegalArgumentException: Parameter 'directory' is not a directory
_details_
java.lang.IllegalArgumentException: Parameter 'directory' is not a directory
at
org.apache.commons.io.FileUtils.validateListFilesParameters(FileUtils.java:545)
at org.apache.commons.io.FileUtils.listFiles(FileUtils.java:521)
at b.g.d(Unknown Source)
at b.a.o(Unknown Source)
at gui.rostereditor.RosterEditorFrame.E(Unknown Source)
at gui.rostereditor.RosterEditorFrame.setVisible(Unknown Source)
at gui.rostereditor.RosterEditorFrame$1.run(Unknown Source)
at java.awt.event.InvocationEvent.dispatch(Unknown Source)
at java.awt.EventQueue.dispatchEventImpl(Unknown Source)
at java.awt.EventQueue.access$200(Unknown Source)
at java.awt.EventQueue$3.run(Unknown Source)
at java.awt.EventQueue$3.run(Unknown Source)
at java.security.AccessController.doPrivileged(Native Method)
at java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
at java.awt.EventQueue.dispatchEvent(Unknown Source)
at java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)
at java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)
at java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)
at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
at java.awt.EventDispatchThread.pumpEvents(Unknown Source)
at java.awt.EventDispatchThread.run(Unknown Source)

**Comment by amis92** [comment]
It seems you have incorrectly setup BattleScribe folder. The program says that directory which it inspects for data files isn't actually a directory in your OS.

Looks like a problem with your BattleScribe setup. Maybe search for help in FAQ/Help on http://www.BattleScribe.net ?

**Comment by HeWhoWatches** [comment]
I was halfway through a reply so I'll post it anyway, but Amis92's advice is correct. Probably best if you double check by clicking the "Manage Data" button. Let's say you wanted to add the 40k data repository. You would paste in the URL http://battlescribedata.appspot.com/repos/wh40k/index.bsi then click the "Add" button, then click "Update Data". 

Depending on what is already showing in the list of URLs, you may need to remove any incorrect entries first by clicking on them, then the "Remove" button. 

The various different links for different game systems can be found here: http://battlescribedata.appspot.com/#/repos - if you hover your mouse over each of the blue titles you should see links ending in ".bsi", similar to the 40k link mentioned earlier. 

Double check you are setting everything up correctly by referring to this help document: http://battlescribe.net/help-gettingstarted-desktop.html

Good luck :)

**Comment by galifrey35** [comment]
ok i deleted battlescribe and it is the latetest version 1.15 the getting
started gui popped up i went to the search for repositories coppied the
address https://github.com/BSData/wh40k. then i went to the add repository
buut put the adrress into the bar where it goes hit add then i clicked the
update data button then i clicked done and got the following error.
- java.lang.IllegalArgumentException: Parameter 'directory' is not a
  directory  Details:java.lang.IllegalArgumentException: Parameter
  'directory' is not a directory    at
  org.apache.commons.io.FileUtils.validateListFilesParameters(FileUtils.java:545)
  at org.apache.commons.io.FileUtils.listFiles(FileUtils.java:521)    at
  b.a.m(Unknown Source)    at b.a.i(Unknown Source)    at gui.a.i.a(Unknown
  Source)    at gui.a.a.c(Unknown Source)    at gui.a.a.c(Unknown Source)
  at gui.a.a$9.actionPerformed(Unknown Source)    at
  javax.swing.AbstractButton.fireActionPerformed(Unknown Source)    at
  javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)    at
  javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source)    at
  javax.swing.DefaultButtonModel.setPressed(Unknown Source)    at
  javax.swing.plaf.basic.BasicButtonListener.mouseReleased(Unknown Source)
  at java.awt.Component.processMouseEvent(Unknown Source)    at
  javax.swing.JComponent.processMouseEvent(Unknown Source)    at
  java.awt.Component.processEvent(Unknown Source)    at
  java.awt.Container.processEvent(Unknown Source)    at
  java.awt.Component.dispatchEventImpl(Unknown Source)    at
  java.awt.Container.dispatchEventImpl(Unknown Source)    at
  java.awt.Component.dispatchEvent(Unknown Source)    at
  java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)    at
  java.awt.LightweightDispatcher.processMouseEvent(Unknown Source)    at
  java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)    at
  java.awt.Container.dispatchEventImpl(Unknown Source)    at
  java.awt.Window.dispatchEventImpl(Unknown Source)    at
  java.awt.Component.dispatchEvent(Unknown Source)    at
  java.awt.EventQueue.dispatchEventImpl(Unknown Source)    at
  java.awt.EventQueue.access$200(Unknown Source)    at
  java.awt.EventQueue$3.run(Unknown Source)    at
  java.awt.EventQueue$3.run(Unknown Source)    at
  java.security.AccessController.doPrivileged(Native Method)    at
  java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
  at java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown
  Source)    at java.awt.EventQueue$4.run(Unknown Source)    at
  java.awt.EventQueue$4.run(Unknown Source)    at
  java.security.AccessController.doPrivileged(Native Method)    at
  java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
  at java.awt.EventQueue.dispatchEvent(Unknown Source)    at
  java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)    at
  java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)    at
  java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)    at
  java.awt.WaitDispatchSupport$2.run(Unknown Source)    at
  java.awt.WaitDispatchSupport$4.run(Unknown Source)    at
  java.security.AccessController.doPrivileged(Native Method)    at
  java.awt.WaitDispatchSupport.enter(Unknown Source)    at
  java.awt.Dialog.show(Unknown Source)    at java.awt.Component.show(Unknown
  Source)    at java.awt.Component.setVisible(Unknown Source)    at
  java.awt.Window.setVisible(Unknown Source)    at
  java.awt.Dialog.setVisible(Unknown Source)    at
  gui.rostereditor.RosterEditorFrame.E(Unknown Source)    at
  gui.rostereditor.RosterEditorFrame.f(Unknown Source)    at
  gui.rostereditor.RosterEditorFrame.i(Unknown Source)    at
  gui.rostereditor.RosterEditorFrame$11.actionPerformed(Unknown Source)    at
  javax.swing.AbstractButton.fireActionPerformed(Unknown Source)    at
  javax.swing.AbstractButton$Handler.actionPerformed(Unknown Source)    at
  javax.swing.DefaultButtonModel.fireActionPerformed(Unknown Source)    at
  javax.swing.DefaultButtonModel.setPressed(Unknown Source)    at
  javax.swing.plaf.basic.BasicButtonListener.mouseReleased(Unknown Source)
  at java.awt.Component.processMouseEvent(Unknown Source)    at
  javax.swing.JComponent.processMouseEvent(Unknown Source)    at
  java.awt.Component.processEvent(Unknown Source)    at
  java.awt.Container.processEvent(Unknown Source)    at
  java.awt.Component.dispatchEventImpl(Unknown Source)    at
  java.awt.Container.dispatchEventImpl(Unknown Source)    at
  java.awt.Component.dispatchEvent(Unknown Source)    at
  java.awt.LightweightDispatcher.retargetMouseEvent(Unknown Source)    at
  java.awt.LightweightDispatcher.processMouseEvent(Unknown Source)    at
  java.awt.LightweightDispatcher.dispatchEvent(Unknown Source)    at
  java.awt.Container.dispatchEventImpl(Unknown Source)    at
  java.awt.Window.dispatchEventImpl(Unknown Source)    at
  java.awt.Component.dispatchEvent(Unknown Source)    at
  java.awt.EventQueue.dispatchEventImpl(Unknown Source)    at
  java.awt.EventQueue.access$200(Unknown Source)    at
  java.awt.EventQueue$3.run(Unknown Source)    at
  java.awt.EventQueue$3.run(Unknown Source)    at
  java.security.AccessController.doPrivileged(Native Method)    at
  java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
  at java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown
  Source)    at java.awt.EventQueue$4.run(Unknown Source)    at
  java.awt.EventQueue$4.run(Unknown Source)    at
  java.security.AccessController.doPrivileged(Native Method)    at
  java.security.ProtectionDomain$1.doIntersectionPrivilege(Unknown Source)
  at java.awt.EventQueue.dispatchEvent(Unknown Source)    at
  java.awt.EventDispatchThread.pumpOneEventForFilters(Unknown Source)    at
  java.awt.EventDispatchThread.pumpEventsForFilter(Unknown Source)    at
  java.awt.EventDispatchThread.pumpEventsForHierarchy(Unknown Source)    at
  java.awt.EventDispatchThread.pumpEvents(Unknown Source)    at
  java.awt.EventDispatchThread.pumpEvents(Unknown Source)    at
  java.awt.EventDispatchThread.run(Unknown Source)*
  *as for going to appspot downloaded the warhammer40k index.bsi ile put it
  into the library and said add from file and it doesn't like that either  *

**Comment by ghost** [comment]
You're using the wrong address for the repo. As @HeWhoWatches noted above, it should be:
http://battlescribedata.appspot.com/repos/wh40k/index.bsi 
not
https://github.com/BSData/wh40k

**Comment by galifrey35** [comment]
ok put that in hit update and got
Message:
Errors occurred updating data files from repositories

> http://battlescribedata.appspot.com/repos/wh40k/index.bsi: Error updating
> from repository http://battlescribedata.appspot.com/repos/wh40k/index.bsi:
> java.net.SocketException: Address family not supported by protocol family:
> 
> ## connect
> 
> Details:
> Errors occurred updating data files from repositories
> http://battlescribedata.appspot.com/repos/wh40k/index.bsi: Error updating
> from repository http://battlescribedata.appspot.com/repos/wh40k/index.bsi:
> java.net.SocketException: Address family not supported by protocol family:
> connect
> at gui.a.t$1.a(Unknown Source)
> at gui.a.t$1.doInBackground(Unknown Source)
> at javax.swing.SwingWorker$1.call(Unknown Source)
> at java.util.concurrent.FutureTask.run(Unknown Source)
> at javax.swing.SwingWorker.run(Unknown Source)
> at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
> at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
> at java.lang.Thread.run(Unknown Source)

**Comment by HeWhoWatches** [comment]
An internet search suggests that the error is caused by your machine using IPv6 rather than IPv4, and Java not using IPv4 by default (see also: http://www.webopedia.com/DidYouKnow/Internet/ipv6_ipv4_difference.html).

These pages suggest a workaround that can be used:
http://stackoverflow.com/questions/16373906/address-family-not-supported-by-protocol-family-socketexception-on-a-specific
http://stackoverflow.com/questions/11850655/how-can-i-disable-ipv6-stack-use-for-ipv4-ips-on-jre
http://mindprod.com/jgloss/javaexe.html#JAVAOPTIONS

Basically, open up a Windows command prompt and run the following:
setx _JAVA_OPTIONS -Djava.net.preferIPv4Stack=true

Then restart the application and try again.

**Comment by amis92** [comment]
@galifrey35 and how is your issue?

**Comment by galifrey35** [comment]
well it's still giving me trouble i had one of the viruses that it said
might cause a problem on stack but i haven't tried the adding that line of
code thing to be honest i'm a bit leary about doing it though.

**Comment by amis92** [comment]
if you need anything further, just reply. In the meantime, I close it here.

**Comment by galifrey35** [comment]
well i downloaded the java update from the battlescribe website and for
some reason it decided to work so i have i'm not looking a gift horse in
 the mouth

On Wed, Jan 28, 2015 at 6:50 PM, Amadeusz Sadowski <notifications@github.com

**Comment by amis92** [comment]
Great! :)
